app/code/Magento/Sales/Model/Order.php

```diff
@@ -708,25 +708,25 @@ public function canCreditmemo()
          * for this we have additional diapason for 0
          * TotalPaid - contains amount, that were not rounded.
          */
-        $totalRefunded = $this->priceCurrency->round($this->getTotalPaid()) - $this->getTotalRefunded();
+        $totalRefundable = $this->priceCurrency->round($this->getTotalPaid()) - $this->getTotalRefunded();
         if (abs((float) $this->getGrandTotal()) < .0001) {
-            return $this->canCreditmemoForZeroTotal($totalRefunded);
+            return $this->canCreditmemoForZeroTotal($totalRefundable);
         }
 
-        return $this->canCreditmemoForZeroTotalRefunded($totalRefunded);
+        return $this->canCreditmemoForZeroTotalRefunded($totalRefundable);
     }
 
     /**
      * Retrieve credit memo for zero total refunded availability.
      *
-     * @param float $totalRefunded
+     * @param float $totalRefundable
      * @return bool
      */
-    private function canCreditmemoForZeroTotalRefunded($totalRefunded)
+    private function canCreditmemoForZeroTotalRefunded($totalRefundable)
```

Comment:
@RikvdHeijden is this method name correct? I noticed you changed the variable, but left the method name intact.

Reply:
@bgorski 

I must admit I do not know the motivation behind the naming and functioning of this code well enough to give any kind of definitive answer. But I can at least share my interpretation.

The logic to determine whether an order can be credited seems to be as follows:

1. If the order is forced to be credited: return that value (presumably true)
2. If an order is in a state where it cannot be credited: return false
3. if the order has a grand total of 0 (or less): determine if the order can be credited by checking the payment status and credit history of the order (1)
4. otherwise: determine if the order can be credited by checking the status of the order (2)

function (1) is called canCreditmemoForZeroTotal which might not be the cleanest name in the world (canCreateCreditmemoWhenGrandTotalIsZero would probably be my pick. But naming is hard) but does seem to be correct.

function (2) - this function - is called canCreditmemoForZeroTotalRefunded. The checks this function does are:
1. Check if there is money left to be refunded
2. Check if the money thats left over after subracting the adjustment fee from the order is still enough to refund the order
3. Check if the order can be edited

if any of those conditions are false, the order cannot be credited. Since the logic of this function is not at all interested in whether the total refunded amount is zero or greater I suppose the method name isn't correct.

I'd suggest a name like
```suggestion
    private function canCreateCreditmemoWhenGrandTotalIsLargerThanZero($totalRefundable)
```
though I suppose that's a rather long name. It might be cleaner to create a single private function called something along the lines of canCreateCreditmemoAccordingToPaymentStatus that combines the grandtotalcheck from canCreditmemo and both functions (1) and (2) since it seems to me that logic is pretty entwined anyway.

I'd love to hear your vision on this.